### PR TITLE
[fix] new dashboard state

### PR DESCRIPTION
**Pull request by graceguo-supercat**
- new dashboard should always be V2, do not show preview message
- According to designer's feedback, if user create new dash/add chart to existed dash from explore view, should show dashboard in view mode.

### CODE REVIEW

File: superset/assets/src/dashboard/reducers/getInitialState.js [hunk 1]
```diff
@@ -18,11 +18,13 @@ import {
 } from '../util/componentTypes';
 
 export default function(bootstrapData) {
+  // ignore editMode flag from server side.
```
Comment (williaster) [inline superset/assets/src/dashboard/reducers/getInitialState.js]: I don't think we should ignore this here, I think it's better for it simply _not_ to be set on explore view side.

**Review by williaster — CHANGES_REQUESTED** [review]
@graceguo-supercat I had one suggestion for the edit mode flag.

also, does this fix the case of "save as"? ie if you "save as" for a v1 or v2 dashboard, is it v2?

File: superset/assets/src/dashboard/reducers/getInitialState.js (hunk 1 above)
Comment (graceguo-supercat) [reply superset/assets/src/dashboard/reducers/getInitialState.js]: old dashboard needs this feature :)
we can remove after transition time.

Comment (williaster) [reply superset/assets/src/dashboard/reducers/getInitialState.js]: @graceguo-supercat why does it need that? can you not add a chart to a v1 dashboard from explorer, then get dumped in a v1 dashboard in view mode?

**Comment by graceguo-supercat** [comment]
can you clarify this case:
owner opens dashboard in V1, he will see promotion message to preview in V2. If non-owner open this dashboard, and tries to save-as, should we save the v1 version in V1 or V2?

**Comment by williaster** [comment]
I think if editing v1, saving should be v1 and if editing v2, saving should be v2.

**Comment by graceguo-supercat** [comment]
Owner can edit or save-as v1 or v2, as long he choose.
if a viewer is not dashboard's owner, he can not edit.

File: superset/assets/src/dashboard/reducers/getInitialState.js (hunk 1 above)
Comment (graceguo-supercat) [reply superset/assets/src/dashboard/reducers/getInitialState.js]: editMode=true flag is set from explore view
https://github.com/apache/incubator-superset/blob/c18ef8903409c20b8526b9acd12d3ba1c4c3664b/superset/assets/src/explore/components/SaveModal.jsx#L111

i just feel it's too many hassle to bring this dashboard logic (preview? version1 ? version2?) to explore view.

Comment (williaster) [reply superset/assets/src/dashboard/reducers/getInitialState.js]: all you have to do is delete `{ edit: 'true' }`,  we discussed that we just never want it to equal true.

File: superset/views/core.py [hunk 2]
```diff
@@ -1562,17 +1564,22 @@ def copy_dash(self, dashboard_id):
...
-                    old_id = value.get('meta').get('chartId')
-                    new_id = old_to_new_sliceids[old_id]
-                    value['meta']['chartId'] = new_id
+            if is_v2_dash:
+                for value in data['positions'].values():
+                    if (
+                        isinstance(value, dict) and value.get('meta') and
+                        value.get('meta').get('chartId')
+                    ):
+                        old_id = '{}'.format(value.get('meta').get('chartId'))
```
Comment (williaster) [inline superset/views/core.py]: @graceguo-supercat can you add a comment about strings vs ints here so the logic is documented somewhere?

**Review by williaster — COMMENTED** [review]
LGTM with one comment request!

**Review by williaster — APPROVED** [review]

File: superset/views/core.py (hunk 2 above)
Comment (graceguo-supercat) [reply superset/views/core.py]: in v2 dashboard position json, chartId is integer type:
```javascript
"DASHBOARD_CHART_TYPE-H10M97Xk-7": {
        "children": [],
        "id": "DASHBOARD_CHART_TYPE-H10M97Xk-7",
        "meta": {
            "chartId": 136,
            "height": 50,
            "sliceName": "Treemap",
            "width": 8
        },
        "type": "DASHBOARD_CHART_TYPE"
    },
```

in older dashboard position json data, slice_id is string type: 
```javascript
   {
        "slice_id": "49", 
        "size_x": 8, 
        "size_y": 16, 
        "v": 1, 
        "col": 33, 
        "row": 24
    }, 
```

**Pull request merged**